src/looprestoration.rs

```diff
@@ -23,3 +24,448 @@ pub union LooprestorationParams {
     pub filter: Align16<[[int16_t; 8]; 2]>,
     pub sgr: LooprestorationParams_sgr,
 }
+
+type pixel = libc::c_void;
+pub type const_left_pixel_row = *const libc::c_void; // *const [pixel; 4]
+
+pub type looprestorationfilter_fn = Option<
+    unsafe extern "C" fn(
+        *mut pixel,
+        ptrdiff_t,
+        const_left_pixel_row,
+        *const pixel,
+        libc::c_int,
+        libc::c_int,
+        *const LooprestorationParams,
+        LrEdgeFlags,
+        libc::c_int,
+    ) -> (),
+>;
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Dav1dLoopRestorationDSPContext {
+    pub wiener: [looprestorationfilter_fn; 2],
+    pub sgr: [looprestorationfilter_fn; 3],
+}
+
+// TODO(randomPoison): Temporarily pub until init fns are deduplicated.
+#[cfg(all(
+    feature = "bitdepth_8",
+    feature = "asm",
+    any(target_arch = "x86", target_arch = "x86_64"),
+))]
+extern "C" {
+    pub(crate) fn dav1d_wiener_filter7_8bpc_sse2(
+        dst: *mut pixel,
```

Comment:
Do we want to use a macro to generate these?  Is that worth it?  It'd look much cleaner and the C version uses macros for this, too.  We can do this in a follow-up PR, though.

Reply:
> Do we want to use a macro to generate these?

IMO it is best to avoid compactifying code using macros. If often interferes with stuff like go-to-definition, grepping for a symbol, and debugging (which is why we need aids like `-Zunpretty=expanded`).

Reply:
Oh, before I saw these comments I went and rewrote this to use a macro 😅

@thedataking I understand your concerns with using macros generally, but in this case I think it's worth it. In addition to making our code more compact, using a macro also has the advantage that it makes it much easier to see which functions all share the same signature, which I think is a big win for readability of the code.